scipy/spatial/transform/rotation.py

```diff
@@ -1465,6 +1466,63 @@ def magnitude(self):
         else:
             return angles
 
+    def mean(self, weights=None, return_variance=False):
+        """Get the mean of the rotation(s).
+
+        Returns
+        -------
+        mean : `Rotation` instance
+            Object containing the mean of the rotations in the current
+            instance.
+        weights : array_like shape (N,), optional
+            Weights describing the relative importance of the rotations. If
+            None (default), then all values in `weights` are assumed to be
+            equal.
+        return_variance : bool, optional
+            Whether to return the variance of the rotation(s). Default is
+            False.
+
+        Notes
+        -----
+        The mean used is the chordal L2 mean (also called the projected or
+        induced arithmetic mean).
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> r = R.random(10, random_state=0)
+        >>> r.mean().as_quat()
+        array([0.96505487, 0.14713643, 0.17081499, 0.13357476])
+        """
+        if weights is None:
+            weights = np.ones(len(self))
+        else:
+            weights = np.asarray(weights)
+            if weights.ndim != 1:
+                raise ValueError("Expected `weights` to be 1 dimensional, got "
+                                 "shape {}.".format(weights.shape))
+            if weights.shape[0] != len(self):
+                raise ValueError("Expected `weights` to have number of values "
+                                 "equal to number of rotations, got "
+                                 "{} values and {} rotations.".format(
+                                    weights.shape[0], len(self)))
+            if np.min(weights) < 0:
```

Comment:
np.any(weights < 0) is slightly more to the point )

Reply:
Nice.